### PR TITLE
[FAB-17529] Reduce and clarify errors in pkg/config

### PR DESCRIPTION
**Pull request by stephyee**
Signed-off-by: Tiffany Harris <tiffany.harris@ibm.com>

#### Type of change

- Improvement (improvement to code, performance, etc)

#### Description

This change focuses on removing/rewording overly descriptive error messages

#### Related issues
[FAB-17529](https://jira.hyperledger.org/browse/FAB-17529)

### CODE REVIEW

File: pkg/config/config_test.go [hunk 1]
```diff
@@ -323,8 +323,8 @@ func TestNewCreateChannelTxFailure(t *testing.T) {
 				profile.Policies = nil
 				return profile
 			},
-			err: errors.New("failed to create default config template: failed to create new channel group: " +
-				"failed to add policies: no policies defined"),
+			err: errors.New("creating default config template: failed to add policies to " +
+				"channel group: no policies defined"),
```
Comment (sykesm) [inline pkg/config/config_test.go:327]: super nit: it's much easier to read and search these when they aren't split across lines.

File: pkg/config/config_test.go [hunk 2]
```diff
@@ -384,8 +384,8 @@ func TestNewCreateChannelTxFailure(t *testing.T) {
 				profile.Policies[ReadersPolicyKey].Rule = "ANYY Readers"
 				return profile
 			},
-			err: errors.New("failed to create default config template: failed to create new channel group: " +
-				"failed to add policies: invalid implicit meta policy rule: 'ANYY Readers' error: unknown rule type " +
+			err: errors.New("creating default config template: failed to add policies to channel group: " +
+				"invalid implicit meta policy rule: 'ANYY Readers': unknown rule type " +
 				"'ANYY', expected ALL, ANY, or MAJORITY"),
```
Comment (sykesm) [inline pkg/config/config_test.go:389]: seems very verbose but not worth changing right now.

**Review by sykesm — APPROVED** [review]
I'm still not sold on the use of "profile" in this code but that's something we can discuss as a follow-on.

File: pkg/config/config_test.go (hunk 1 above)
Comment (stephyee) [reply pkg/config/config_test.go:327]: Fair enough, I've been splitting these on the suggestion by the linter.

File: pkg/config/config_test.go (hunk 2 above)
Comment (stephyee) [reply pkg/config/config_test.go:389]: It's been trimmed down some. Is it reasonable for the consumer to know which rules are valid?

**Comment by stephyee** [comment]
Agreed - there is a follow up work to rename profile to something more appropriate.

**Pull request merged**